### PR TITLE
Container version

### PR DESCRIPTION
**Pull request by aethr**
## Description

Make the openmethane package version available inside the container so that data files can be annotated with the version that was used to process them.

## Checklist

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Changelog item added to `changelog/`

### CODE REVIEW

**Comment by aethr** [comment]
@prayner the container will now have access to the `OPENMETHANE_VERSION` env var when running shell or python scripts. Where do you think the best place would be to embed the package version so that it flows through to our result data files?

This container is used for several jobs:
- CMAQ fetching and preprocessing
- Fetch and process TROPOMI data
- Bias correction
- fourdvar daily run
- fourdvar monthly run

At a minimum, I'd say we should write the version:
- fourdvar daily run
- fourdvar monthly run

I believe both of these make use of the `archive` method, so this might be a logical place for it:    - possibly in the `archive` method? https://github.com/openmethane/openmethane/blob/d0b65742a5aa3421e7af29d13d2053f0661fe09d/src/fourdvar/datadef/abstract/_physical_abstract_data.py#L123-L127

File: src/fourdvar/datadef/abstract/_physical_abstract_data.py [hunk 1]
```diff
@@ -124,6 +124,7 @@ def unc(spc):
             "SDATE": np.int32(dt.replace_date("<YYYYDDD>", date_defn.start_date)),
             "EDATE": np.int32(dt.replace_date("<YYYYDDD>", date_defn.end_date)),
             "VAR-LIST": "".join([f"{s:<16}" for s in self.spcs]),
+            "openmethane_version": os.getenv("OPENMETHANE_VERSION", "development"),
```
Comment (aethr) [inline src/fourdvar/datadef/abstract/_physical_abstract_data.py:127]: Using snake case for `openmethane_version` contrasts the existing attributes which are ALL-CAPS. However, in trying to follow CF Conventions for new attributes, I've adopted their style which is lowercase, with _ as a separator.

https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#attribute-appendix

**Comment by prayner** [comment]
I just checked that calculate_average_emissions in postproc copies
attributes from the dataset for the prior to the output dataset.
However the prior.nc file is produced by make_prior.py which doesn't
use PhysicalAbstractData so the environment variable will need to be
written explicitly into the attributes of the root group. It should
also get written as a key in the domain definition at the start of the
obs list produced by tropomi_methane_preprocess .

**Comment by aethr** [comment]
> However the prior.nc file is produced by make_prior.py which doesn't use PhysicalAbstractData so the environment variable will need to be written explicitly into the attributes of the root group.

It looks as though the `setup` method in `user_driver.py` does load `prior.ncf` using the `PhysicalData` class and then writes it back to disk: 
https://github.com/openmethane/openmethane/blob/d6d30aa745e10dc67e28dd2897c600fc85db3139/src/fourdvar/user_driver.py#L49-L51

Sure enough, I'm seeing the `openmethane_prior` attribute when I view it with `ncdump` after doing a local run.

>  It should also get written as a key in the domain definition at the start of the obs list produced by tropomi_methane_preprocess .

I'll look at adding this.

**Comment by aethr** [comment]
> It should also get written as a key in the domain definition at the start of the obs list produced by tropomi_methane_preprocess .

I've added it to `get_domain` within `obs_preprocess.model_space` which I believe is where the domain definition is created.

```
    def get_domain(self):
        domain = deepcopy(self.gridmeta)
        domain["SDATE"] = self.sdate
        domain["EDATE"] = self.edate
        domain["openmethane_version"] = os.getenv('OPENMETHANE_VERSION')
```

**Review by prayner — APPROVED** [review]

**Pull request merged**